### DISCUSSION TITLE
Powershell script to split catalog .csv file into parts for Azure DevOps import

### DISCUSSION THREAD

**Discussion by FH-Inway** [microsoft/dynamics365patternspractices#143] [Show and tell]
The following link describes how the business process catalog can be imported as work items into Azure DevOps:
https://learn.microsoft.com/en-us/dynamics365/guidance/business-processes/about-import-catalog-devops

This involves multiple manual steps that make it a bit cumbersome to set this up. One of the steps is to split the .csv files into multiple files, because the Azure DevOps import only allows files of up to 1000 rows.

I (well, mostly Github Copilot) wrote a script to make that step a bit easier. You can download the script from my fork of the https://github.com/MicrosoftDocs/dynamics365-guidance repository:
https://github.com/FH-Inway/dynamics365-guidance/blob/experiments/tools/business-process-catalog-for-azure-devops/Split-CatalogCSVForDevOpsImport.ps1

If you have a Github account, you can instead create a codespace on the [experiments](https://github.com/FH-Inway/dynamics365-guidance/tree/experiments) branch. The codespace is already preconfigured with everything needed to start experimenting with the script. Just open the [test.ps1](https://github.com/FH-Inway/dynamics365-guidance/blob/experiments/tools/business-process-catalog-for-azure-devops/test.ps1) file and run it with F5 or the run button. It will create the split files in the root folder.

Note that you can edit the .csv file before running the script (e.g. to remove columns you don't need). The split logic only requires the "Work Item Type" column.

# Create codespace
![Create a codespace](https://github.com/microsoft/dynamics365patternspractices/assets/33372796/12ba9cb6-b1cf-413f-a4c0-ec86bb2f650f)

# Run the script
![Run the test.ps1 script](https://github.com/microsoft/dynamics365patternspractices/assets/33372796/66e32da3-8e31-4a7f-a23a-efd175ab107c)

# See the result
![Split files](https://github.com/microsoft/dynamics365patternspractices/assets/33372796/681f7194-cf07-4179-bef1-d90ab1444619)

**Comment by rachel-profitt** [comment]
Thanks for contributing this, I am interested to see if others find this useful and if we should make a process like this standard.
